src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -435,19 +441,22 @@ def _to_data_or_iter(
         to_pandas: bool = False,
         to_iter: bool = False,
         num_statements: Optional[int] = None,
+        fetch_pandas_kwargs: Optional[Dict[str, Any]] = None,
     ) -> Dict[str, Any]:
         if to_pandas:
+            fetch_pandas_kwargs = fetch_pandas_kwargs or {}
             try:
                 data_or_iter = (
                     map(
                         functools.partial(
                             _fix_pandas_df_fixed_type, results_cursor=results_cursor
                         ),
-                        results_cursor.fetch_pandas_batches(),
+                        results_cursor.fetch_pandas_batches(**fetch_pandas_kwargs),
```

Comment:
ML team suggests we use `split_blocks=True` instead of doing parameter passing from client to pyarrow. This is because our expectation is that memory issues with default value `split_blocks=False` will be fixed soon, so they don't want to change their code first to set it to true and then remove it.

What does the team think about this?

More discussion on this topic is done here: https://docs.google.com/document/d/1vz1891wBIayuNdR351VUWONozUc_ErJBSBrdAEuhosg/edit#heading=h.4as47gcvtnu5